zebra-utils/src/bin/search-issue-refs/main.rs

```diff
@@ -0,0 +1,223 @@
+//!
+//! Example usage:
+//!
+//! (from the root directory of the Zebra repo)
+//! ```console
+//! GITHUB_TOKEN={valid_github_access_token} search-issue-refs
+//! ```
+//!
+//! Example output:
+//!
+//! > Found 3 possible issue refs, checking Github issue statuses..
+//! >
+//! > --------------------------------------
+//! > Found reference to closed issue #1140: ./zebra-test/tests/command.rs:130:57
+//! > <https://github.com/ZcashFoundation/zebra/issues/1140>
+//! >
+//! > --------------------------------------
+//! > Found reference to closed issue #1140: ./zebra-test/tests/command.rs:157:51
+//! > <https://github.com/ZcashFoundation/zebra/issues/1140>
+//! >
+//! > --------------------------------------
+//! > Found reference to closed issue #1140: ./zebra-test/tests/command.rs:445:61
+//! > <https://github.com/ZcashFoundation/zebra/issues/1140>
+//! >
+//! > Found 3 references to closed issues.
+
+use std::{
+    env,
+    ffi::OsStr,
+    fs::{self, File},
+    io::{self, BufRead},
+    path::{Path, PathBuf},
+};
+
+use color_eyre::eyre::Result;
+use regex::Regex;
+use reqwest::{
+    header::{self, HeaderMap, HeaderValue},
+    ClientBuilder,
+};
+
+use tokio::task::JoinSet;
+use zebra_utils::init_tracing;
+
+const GITHUB_TOKEN_ENV_KEY: &str = "GITHUB_TOKEN";
+
+const VALID_EXTENSIONS: [&str; 4] = ["rs", "yml", "yaml", "toml"];
+
+fn check_file_ext_and_path(ext: &OsStr, path: &Path) -> bool {
+    VALID_EXTENSIONS
+        .into_iter()
+        .any(|valid_extension| valid_extension == ext)
+        && !path.starts_with("/target/")
+}
+
+fn search_directory(path: &PathBuf) -> Result<Vec<PathBuf>> {
+    Ok(fs::read_dir(path)?
+        .filter_map(|entry| {
+            let path = entry.ok()?.path();
+
+            if path.is_dir() {
+                search_directory(&path).ok()
+            } else if path.is_file() {
+                match path.extension() {
+                    Some(ext) if check_file_ext_and_path(ext, &path) => Some(vec![path]),
+                    _ => None,
+                }
+            } else {
+                None
+            }
+        })
+        .flatten()
+        .collect())
+}
+
+fn github_issue_url(issue_id: &str) -> String {
+    format!("https://github.com/ZcashFoundation/zebra/issues/{issue_id}")
+}
+
+fn github_issue_api_url(issue_id: &str) -> String {
+    format!("https://api.github.com/repos/ZcashFoundation/zebra/issues/{issue_id}")
+}
+
+#[derive(Debug)]
+struct PossibleIssueRef {
+    file_path: PathBuf,
+    line_number: usize,
+    column: usize,
+    id: String,
+}
+
+/// Process entry point for `search-issue-refs`
+#[allow(clippy::print_stdout, clippy::print_stderr)]
+#[tokio::main]
+async fn main() -> Result<()> {
+    init_tracing();
+    color_eyre::install()?;
+
+    let file_paths = search_directory(&".".into())?;
+
+    let issue_regex =
+        Regex::new(r"(https://github.com/ZcashFoundation/zebra/issues/|#)\d{4}").unwrap();
```

Comment:
I think this regex won't match strings with less than 4 digits, for example 
`#42` or `#420`, which are valid issue labels.

Maybe we could try something like this?
```suggestion
        Regex::new(r"(https://github.com/ZcashFoundation/zebra/issues/|#)\d+").unwrap();
```

However, matches of variable length will collide with the code at https://github.com/zcashfoundation/zebra/blob/821919f9e03c4536d7a7a59d0b76144429e1b4f3/zebra-utils/src/bin/search-issue-refs/main.rs#L120.

Reply:
Great point! Updated to check for 1-4 digits in [`b6439dd`](https://github.com/ZcashFoundation/zebra/pull/6347/commits/b6439ddf51fefb6c0b691cc95da8bec9965f3d3d).